### PR TITLE
Added cfg_file command line args

### PR DESCRIPTION
**Pull request by sebhz**
Let's use this arg parser.
Other args that would be nice to have are --fullscreen and --autorun_cmd (in answer to issue #38). Both those would allow to use caprice32 as a screensaver which would be cool...

### CODE REVIEW

File: src/argparse.cpp [hunk 1]
```diff
@@ -2,15 +2,21 @@
...
-   {"version", no_argument, nullptr, 'V'},
-   {"help",    no_argument, nullptr, 'h'},
-   {nullptr, 0,  nullptr, 0},
+   {"cfg_file", required_argument, nullptr, 'c'},
+   {"version",  no_argument, nullptr, 'V'},
+   {"help",     no_argument, nullptr, 'h'},
+   {nullptr, 0, nullptr, 0},
 };
 
+void initArgs(struct capriceArgs& args)
```
Comment (ColinPitrat) [inline src/argparse.cpp]: The initArgs function is not needed: the default constructor of the string will set it to an empty string.
If this wasn't the case (or if we wanted to use different default values), making capriceArgs a class with a constructor would be better (no risk of forgetting to call init).

File: src/cap32.cpp [hunk 2]
```diff
@@ -3347,7 +3348,7 @@ int cap32_main (int argc, char **argv)
       exit(-1);
    }
 
-   loadConfiguration(CPC, getConfigurationFilename()); // retrieve the emulator configuration
+   loadConfiguration(CPC, args.cfgFilePath != "" ? args.cfgFilePath : getConfigurationFilename()); // retrieve the emulator configuration
```
Comment (ColinPitrat) [inline src/cap32.cpp]: I'd prefer args.cfgFilePath.empty() ? ...
Also, how does it behave if the file doesn't exist (If I remember well, it now works but I'm not 100% sure) ?

Comment (sebhz) [inline src/cap32.cpp]: Fixed. And yes it does work well if the file does not exist: no error reported, and you get a default configuration.

File: src/argparse.cpp (hunk 1 above)
Comment (sebhz) [inline src/argparse.cpp]: At least I'll learn quite a bit about C++ when playing around with Caprice32 :-)
The proper way as you mention would be to make capriceArgs a class with a constructor. Some other possible common arguments (e.g. --fullscreen) would be boolean and need initialization. Other might be strings that we want to initialize to some non-empty values. I'll find some time to create the class in the next days and will update this pull request with it.

**Comment by sebhz** [comment]
Turns out it was pretty simple after all... I put the class CapriceArgs class definition in argparse.h, and did not create a namespace for it. In the end it is mostly meant to be a collection of members, so I did not feel it was required. I also made args a global variable of cap32.cpp which makes it easier to pass around - since some args might be used all over the place.

**Comment by ColinPitrat** [comment]
Thanks for this. I didn't think of using Caprice32 as a screensaver but indeed, some demo would make good ones !

**Comment by ColinPitrat** [comment]
For info, I implemented the -a/--autocmd option that allows to automatically execute one (or multiple if repeated) command(s).

**Comment by 2play** [comment]
@ColinPitrat Hello! Im very happy to see issues resolved and your emu is being updates.
I m using a Pi3B
I tried to apply a custom cfg to the path but does not work
--cfg_file=/opt/retropie/configs/amstradcpc/cap32.cfg

Also what would be the correct command with -a to autoload the game (as on lr-carpice32 types RUN"GAMENAME no extension so zip boots)

on my pi try this:
`caprice32 = "/opt/retropie/emulators/caprice32/cap32 -a %ROM%"
caprice32-cfg = "/opt/retropie/emulators/caprice32/cap32 --cfg_file=/opt/retropie/configs/amstradcpc/cap32.cfg %ROM%"
`

Is the --fullscreen or similar command line option implemented?

Tx in advance for your kind reply

**Comment by ColinPitrat** [comment]
Just to be sure you're really using this cap32, can you try the following?
`./cap32 --version`

What does not work when providing a custom config file? Do you have an error message? What do you see on the console? (there should be a 'Using configuration file:' message).

The command to autoload the game depend on the game itself, but basically the following:
`./cap32 game.dsk -a 'run"game'`
Providing `run"game` in the emulator starts the game.

There's no --fullscreen, the idea is to put everything in the config file and keep command line options minimal. It's always kind of complex to mix command line options and configuration.
One thing you can use is:
`./cap32 -a CAP32_FULLSCRN`
which will, if you start in window mode, switch to fullscreen shortly after.

**Comment by 2play** [comment]
impressed! very quick reply

here is my version whic I compiled from source
Caprice32 v4.5.0-b7797237d17e9c88f5a8e8bdfc3dc298d1a66b2d
Compiled with: HAVE_GL

fullscreen in cfg is nice option.
The default config on pi will be
`caprice32 = "/opt/retropie/emulators/caprice32/cap32 -a %ROM%"`
i tried to load for ecample Airwolf (Europe).zip
which brings me to the basic. if i type then cat and then RUN"AIRWOLF it loads

I created as above an entry to the emulators.cfg (where the commands to load a game used)
`caprice32-cfg = "/opt/retropie/emulators/caprice32/cap32 --cfg_file=/opt/retropie/configs/amstradcpc/cap32.cfg %ROM%"`

Ok from cli seems that it recognizes the custom cfg but not from above

" ./cap32 --cfg_file=/opt/retropie/configs/amstradcpc/cap32.cfg /home/pi/RetroPie/roms/amstradcpc/Airwolf\ \(Europe\).zip
Using configuration file: /opt/retropie/configs/amstradcpc/cap32.cfg
Init of video subsystem failed: Unable to open a console terminal
video_init() failed. Aborting.
"
Ignore error as i ve run from ssh

Tried
-a CAP32_FULLSCRN not working when used 
`caprice32 = "/opt/retropie/emulators/caprice32/cap32 -a CAP32_FULLSCRN %ROM%"`

Tired
`caprice32 = "/opt/retropie/emulators/caprice32/cap32 -a cat -'run"%ROM%'"`
or 
`caprice32 = "/opt/retropie/emulators/caprice32/cap32 -a cat -run/"%ROM%"`
no go

as said im using .zip files. is the cap32 working with only direct *.dsk files?

**Comment by 2play** [comment]
From cli I found that it was looking 6128.rom in /usr/local/share
I made a symbolic link and it solved.
but no games load automatically either or from cli

**Comment by 2play** [comment]
custom cfg files works fine now. to make sure its correct i played with on off fps!
so this is good now

**Comment by 2play** [comment]
I noticed in cfg 
`# scr_window
#   0: Fullscreen mode
#   1: Windowed mode
scr_window=1`

after setting 
# scr_style
#   0: Half size with hardware flip
#   1: Double size with hardware flip
#   2: Half size
#   3: Double size
#   4: Super eagle
#   5: Scale2x
#   6: Advanced Scale2x
#   7: TV 2x
#   8: Software bilinear
#   9: Software bicubic
#  10: Dot matrix
#  11: OpenGL scaling
scr_style=11

(to 11 it wouldnt boot, back to 1 and now its full size window)

**Comment by ColinPitrat** [comment]
Excellent, so everything works in the end?

**Comment by 2play** [comment]
Yes only thing left is to make the autorun work. 

quick one. why the 6128+ systems give the F1 and F2 Burning Rubber option?
How the gx4000 will boot with this?

**Comment by 2play** [comment]
If I may suggest you should colaborate with KaosOverride
He also has a fork or Caprice https://github.com/KaosOverride/CapriceRPI
and auto load is smooth and menu.

I think a great new caprice will be born with all systems supported and all nice features.
Just a kind suggestion as he is missing the 6128+ support  etc

**Comment by ColinPitrat** [comment]
The 6128+ like the GX4000 boots on the cartridge by default.
The default cartridge of the 6128+ has Amstrad Basic & Burnin' Rubber with this F1/F2 menu. If you use a different cartridge, it will directly boot on the game. If you were to use a 6128+ default cartridge with a GX4000, it would also show you the menu and you wouldn't be able to do anything as there is no keyboard.

**Comment by 2play** [comment]
About the CapriceRPI i can see he has update the src cpe32.cpp 3d ago. Now in version 1.3d
I have tested the c and planning to test d

About gx4000 i ll test above. 
Im using `caprice32 = "/opt/retropie/emulators/caprice32/cap32 --cfg_file=/opt/retropie/configs/gx4000/cap32.cfg %ROM%"`
where i ve set the model to 3 correct?
# model
#   0: CPC464
#   1: CPC664
#   2: CPC6128
#   3: CPC6128+
model=3

**Comment by ColinPitrat** [comment]
GX4000 doesn't support disks (.dsk) it only has cartridge port. The dsk won't autoload, you have to pass the command to launch the game as a -a option if you don't want to have to type it once in Amstrad Basic prompt.

**Comment by 2play** [comment]
yes i get that. i had a set in .bin but wont load. cpr works smooth as you said

now my final thing to check is normal amstrad can load with -a the dsk :-)
tx for your time and replies @ColinPitrat

**Pull request merged**